simplemma/strategies/dictionaries/trie_directory_factory.py

```diff
@@ -0,0 +1,161 @@
+import hashlib
+import logging
+from collections.abc import MutableMapping
+from functools import lru_cache
+from pathlib import Path
+from typing import ByteString, Dict, List, Optional, cast
+
+from marisa_trie import BytesTrie, HUGE_CACHE  # type: ignore[import-not-found]
+from platformdirs import user_cache_dir
+
+from simplemma.strategies.dictionaries.dictionary_factory import (
+    SUPPORTED_LANGUAGES,
+    DictionaryFactory,
+    _load_dictionary_from_disk,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class TrieWrapDict(MutableMapping):
+    """Wrapper around BytesTrie to make them behave like dicts."""
+
+    def __init__(self, trie: BytesTrie):
+        self._trie = trie
+
+    def __getitem__(self, item):
+        return self._trie[item.decode()][0]
```

Comment:
Having to do this decoding after just encoding feels a bit wasteful, every time makes me think if we did a right change changing this to bytestring...

Reply:
While I somewhat agree, it wasn't much better before, as `BytesTrie` expects the keys to be strings and the values to be bytestrings. So while now the keys have to be encoded/decoded, before the values had to be.